lib/beaker/hypervisor/fusion.rb

```diff
@@ -54,12 +55,65 @@ def provision
         end
         time = Time.now - start
         @logger.notify "Spent %.2f seconds resuming VM" % time
+
+        begin
+          try_ssh_connection(host.name, host['user'], host['ssh'])
```

Comment:
I'm not a big fan of moving ssh connection knowledge outside of the ssh connection module.  Might be more reasonable to update the ssh_connection.connect to take a parameter to reduce the number of re-tries and then use it instead.  The, instead of this whole block, you could just run a random command on the host (like: ```on host, 'echo'```) to establish a connection.  If the connection fails to establish catch the runtime error and raise from there, otherwise you can then check the ip address.